projects/storefrontapp-e2e-cypress/cypress/integration/regression/my-account/order-cancellation.e2e-spec.ts

```diff
@@ -0,0 +1,39 @@
+import { loginSuccessfully } from '../../../helpers/checkout-as-persistent-user';
+import {
+  cancelOrder,
+  placeOrderAndVerifyHistory,
+} from '../../../helpers/order-cancellation';
+
+let orderCode: string;
+
+context('Order Cancellation - Desktop', () => {
+  before(() => {
+    cy.window().then((win: Window) => {
+      win.localStorage.clear();
+    });
+  });
+
+  describe('Order Placement', () => {
+    beforeEach(() => {
+      cy.restoreLocalStorage();
+    });
+
+    afterEach(() => {
+      cy.saveLocalStorage();
+    });
+
+    it('should login and redirect to front page', () => {
+      loginSuccessfully();
+    });
+
+    it('should place an order and then display order history details page', () => {
+      orderCode = placeOrderAndVerifyHistory();
```

Comment:
I can take out the verification of order history and add a check to make sure the order's status is pending.

I'm not saying that the order placed is taking forever, but rather that it is taking a long time because it waits for it to display in the user's order history. 

Look carefully at the `placeOrderAndVerifyHistory` method's code: it's already using the `waitForOrderToBePlacedRequest` method. 

I will add the additional cases, but can you elaborate on the difference between consignment and unconsignment entry?

Reply:
`I can take out the verification of order history and add a check to make sure the order's status is pending.`

good

```I'm not saying that the order placed is taking forever, but rather that it is taking a long time because it waits for it to display in the user's order history.

Look carefully at the placeOrderAndVerifyHistory method's code: it's already using the waitForOrderToBePlacedRequest method.```

Looking carefully, at the placeOrderAndVerifyHistory method, you call `waitForOrderToBePlacedRequest ` and `waitForOrderWithConsignmentToBePlacedRequest`, which first waits until the order exists in the response of the order before asserting and doing the other wait. Basically, you are waiting for the order to be placed as in successfully saving it in the system, which will return the order when requesting for it, so that when you visit the order history page, the order history exists. (I wrote the functionality, so I am pretty sure that's right and the consignment is a derivation of it since Marcin needed it when we had a bug with unconsigned/consigned orders in the order detail pages, which is now fix)

Furthermore, in your function, it waits for the consignment to at least be in the response afterwards. However this is redundant, since you only want to `place order and verify the order history`, not the order details. This redundancy elongated the function as it was additive to the previous wait which kept `re-trying`, therefore taking `a long time`.

```I will add the additional cases, but can you elaborate on the difference between consignment and unconsignment entry?```

Difference between unconsignment and consignment are unconsignment was not fixed yet by the backend (everything is converted under the hood in the backend). It will become consigned only when the system deems that it's ready for to be ready for the person, for example, when it's being delivered, it will have different display status.  

`In transit, waiting, delivering...` (consigned) 
`pending, created... pending...` (unconsigned)

look at it as if you bought something, and the system is processing your information. It will turn to consignments once it gets delivered or so.